### PR TITLE
[release/2.6] [SWDEV-535259] enable miopen channels last 3d for conv and batchnorm

### PR DESCRIPTION
**Pull request by jeffdaily**
The same env vars PYTORCH_MIOPEN_SUGGEST_NHWC and PYTORCH_MIOPEN_SUGGEST_NHWC_BATCHNORM now also enable channels last 3d in addition to 2d.

### CODE REVIEW

File: aten/src/ATen/native/Normalization.cpp [hunk 1]
```diff
@@ -539,6 +539,7 @@ BatchNormBackend _select_batch_norm_backend(
       && (input.suggest_memory_format() == MemoryFormat::Contiguous
 #if (defined(USE_ROCM) && ROCM_VERSION >= 60500)
         || (input.suggest_memory_format() == MemoryFormat::ChannelsLast && PYTORCH_MIOPEN_SUGGEST_NHWC_BATCHNORM)
+        || (input.suggest_memory_format() == MemoryFormat::ChannelsLast3d && PYTORCH_MIOPEN_SUGGEST_NHWC_BATCHNORM)
```
Comment (BrianHarrisonAMD) [inline aten/src/ATen/native/Normalization.cpp:542]: Does this need 2D & 3D as well?

Looks like it used to be 2d, and now it's only 3d.

**Comment by BrianHarrisonAMD** [comment]
I think we also need to change this file:
- https://github.com/ROCm/pytorch/blob/0a6e1d6b9bf78d690a812e4334939e7701bfa794/aten/src/ATen/native/ConvUtils.h#L355

or else it won't try to use NHWC for MIOpen convs even with flag set.

**Comment by BrianHarrisonAMD** [comment]
Not sure on the usual process, but we might need to merge this into 2.6 & 2.7 once ready.

**Comment by jeffdaily** [comment]
> I think we also need to change this file to update the NHWC check:
> 
> * https://github.com/ROCm/pytorch/blob/0a6e1d6b9bf78d690a812e4334939e7701bfa794/aten/src/ATen/native/ConvUtils.h#L378
> 
> or else it won't try to use NHWC for MIOpen convs even with flag set it looks like.

Not so, a few lines later the return statement does an or of this bool var and the 2d bool var which means 3d is enabled if 2d is enabled.  All the other changes in this PR were the lines actually preventing channels last 3d.

**Comment by jeffdaily** [comment]
!cherry-pick --onto release/2.7

**Comment by okakarpa** [comment]
Created branch [autogenerated/release/2.7_cherry-pick_pr-2209](https://github.com/ROCm/pytorch/tree/autogenerated/release/2.7_cherry-pick_pr-2209) and https://github.com/ROCm/pytorch/pull/2232

**Comment by jithunnair-amd** [comment]
!cherry-pick --onto rocm7.0_internal_testing
